File: FE/src/types/product.ts

export interface ProductResponse {
    next: null;
    previous: null;
    results: Product[];
    total: number;
    page: number;
}

export interface Product {
    id: number;
    name: string;
    slug: string;
    get_absolute_url: string;
    description: string;
    price: string;
    image: string;
    discount: string;
    category: string;
    tags: string[];
    get_image: string;
}

File: my_app/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String slug;
  final String description;
  final String price;
  final String imageUrl;
  final List<String> tags;

  Product({
    required this.id,
    required this.name,
    required this.slug,
    required this.description,
    required this.price,
    required this.imageUrl,
    this.tags = const [],
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
      description: json['description'],
      price: json['price'],
      imageUrl: json['get_image'],
      tags: [],
    );
  }
}
